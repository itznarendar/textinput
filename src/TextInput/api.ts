export type InputPrtext = Omit<
  React.InputHTMLAttributes<HTMLInputElement>,
  'onChange'
> & {
  label?: React.ReactNode
  /** css class for wrapper, add `text-text-input-with-state` for showing status with input control */
  className?: string
  onChange?: (
    value: string,
    event?: React.ChangeEvent<HTMLInputElement>
  ) => void

  /**
   * validate if the input text has error or not
   */
  validate?: 'ok' | 'error'

  /**
   * show close button to clear all the text inside input
   */
  clearAll?: boolean
  /**
   * Callback when clearall is clicked
   */
  onClearAll?: Function
  ref?: React.Ref<HTMLInputElement>
  /**
   * show copy button to copy the text inside input
   */
  copy?: boolean
  /**
   * show help text for input field
   */
  children?: React.ReactNode

  errors?: React.ReactNode
}

export type LabelPrtext = {
  label: React.ReactNode
  inputId?: string
  required?: boolean
  validate?: InputPrtext['validate']
}

export type IpAddress = {
  first: string
  second: string
  third: string
  fourth: string
}
