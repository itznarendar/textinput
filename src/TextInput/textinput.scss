@import '../Typography/typography.scss';

$font-icon-close: '\e81a';

.text-text-input {
  display: flex;
  flex-direction: column;

  &:focus {
    outline: none;
  }
  input {
    display: block;
    width: 100%;
    border-radius: 0.25rem;
    border: 1px solid var(--color-elevation-40);
    padding: 8px 14px;
    background-color: var(--color-elevation-20);
    border-color: var(--color-elevation-40);
    color: var(--color-grey-800) !important;

    &:hover:not(:disabled) {
      outline: none;
      border: 1px solid var(--color-grey-600);
    }
    &:focus {
      outline: none;
      border: 1px solid var(--color-primary);
    }
  }
  label {
    color: var(--color-grey-600);
    display: block;
    margin-bottom: 4px;
  }

  .text-label-required::after {
    content: '*';
    margin: 0 3px;
    color: var(--color-red);
  }

  input:focus + label {
    color: var(--color-primary);
  }

  input:disabled {
    background-color: var(--color-grey-100);
  }

  &:focus-within .counter {
    visibility: visible;
  }

  &:focus-within .inputIcon {
    visibility: visible;
  }

  .label-error {
    color: var(--color-red);
  }

  .text-input-error {
    border-color: var(--color-red);
  }
  .textinput-action-button {
    font-size: inherit;
    position: absolute;
    right: 0.4rem;
    top: 50%;
    transform: translateY(-50%);
    // visibility: hidden;
    .inputIcon {
      font-size: inherit;
    }
  }

  .validateIcon {
    position: absolute;
    right: 4px;
    top: 50%;
    transform: translateY(-50%);
  }

  &:focus-within .validateIcon {
    visibility: hidden;
  }
}

.footer {
  padding: 4px 14px;
  color: var(--color-red);
}

.counter {
  position: absolute;
  right: 0;
  margin-top: 2px;
  visibility: hidden;
}

.text-text-input + .text-text-input {
  padding-top: 20px;
}

.text-text-input .dot::after {
  content: '';
  font-size: 18px;
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  right: 0;
  border: 1px solid;
  border-radius: 100%;
  width: 4px;
  height: 4px;
  background: black;
}

.text-text-input input.text::-webkit-outer-spin-button,
.text-text-input input.text::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.text-text-input input.text {
  -moz-appearance: textfield;
}

.text-input-wrapper {
  display: flex;
  flex-direction: column;
}

.text-input-inner {
  position: relative;
}

.text-input-outer {
  border: 1px solid var(--color-elevation-40);
  border-radius: 0.25rem;

  &:focus-within {
    border-color: var(--color-primary);
  }

  &:hover {
    border-color: var(--color-grey-600);
  }
}


.text-input-outer:focus-within + label,
.text-input-inner:focus-within + label {
  color: var(--color-primary);
}

.text-text-input.text-text-input-with-state {
  input {
    padding-left: 2rem;
  }

  .text-text-input-content .icon-status,
  .text-text-input-content .test-spinner {
    position: absolute;
    left: 0.25rem;
    z-index: 1;
    top: 0.5rem;
  }

  .text-text-input-content .icon-status {
    left: 0.375rem;
    top: 0.625rem;
  }

  .text-text-input-label + .text-text-input-content {
    .test-spinner {
      top: 1.875rem;
    }
    .icon-status {
      top: 2rem;
    }
  }
}

/**
* Tag Input Type
*/
.text-text-input {
  .react-tag-input {
    border: none;
    padding: .1875em .5em;
    background-color: var(--color-elevation-20);
  }
  
  .react-tag-input__tag {
    background: none;
    border-radius: 20px;
    border-width: 1px;

    .react-tag-input__tag__content {
      @extend .body-3;
      padding: 0 0 0 0.625rem;
      color: var(--color-grey-800)
    }

    .react-tag-input__tag__remove {
      background: none;
      color: var(--color-grey-600);

      &::before {
        transform: none;
        font-family: "fontello";
        content: $font-icon-close;
        background:none;
        top: 30%;
        left: 30%;
      }

      &::after {
        transform: none;
        background: none;
      }
    }
  }

  input.react-tag-input__input {
    @extend .body-2;
    display: inline;
    width: auto;
    margin: 0;
    padding: 0.5em 0.375em;
    border: none;

    &:focus,
    &:hover {
      border: none;
    }
  }
}
