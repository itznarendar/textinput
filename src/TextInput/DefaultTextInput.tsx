import React, { useRef, useState, useEffect } from 'react'
import clsx from 'clsx'
import { InputPrtext } from './api'
import Button from '../Button'
import { useTranslation } from 'react-i18next'

function DefaultTextInput({
  maxLength,
  clearAll,
  validate,
  type,
  className,
  copy,
  onChange,
  onClearAll,
  ...rest
}: InputPrtext) {
  const { t } = useTranslation()
  const inputRef = useRef<HTMLInputElement | null>(null)
  const [confirmation, setConfirmation] = useState(false)
  const [value, setValue] = useState(rest.defaultValue || rest.value)

  // Update value state whenerver value prtext change
  useEffect(() => {
    setValue(rest.value)
  }, [rest.value])

  // Hide the tooltip after clicking on copy icon in 2.5s
  useEffect(() => {
    let timerId: NodeJS.Timeout

    if (confirmation) {
      timerId = setTimeout(() => {
        setConfirmation(false)
      }, 2500)
    }

    return () => {
      clearTimeout(timerId)
    }
  }, [confirmation])

  function clear() {
    if (inputRef.current) {
      inputRef.current.value = ''
      onChange?.('')
      setValue('')
      onClearAll?.()
    }
  }

  function renderLengthMessage() {
    if (maxLength) {
      return value ? (
        <span className="counter body-3 test-grey-800">
          {`${String(value).length} / ${maxLength}`}
        </span>
      ) : null
    }

    return null
  }

  function renderClearButton() {
    if (clearAll && value) {
      return (
        <Button
          variant="icon"
          type="button"
          onClick={clear}
          className="textinput-action-button"
        >
          <i className="icon-close inputIcon" />
        </Button>
      )
    }
    return null
  }

  function renderHelperText() {
    if (validate) {
      return validate === 'error' ? (
        <i className="icon-error-circle text-lg validateIcon text-text-red" />
      ) : (
        <i className="icon-check text-lg validateIcon text-text-green" />
      )
    }

    return null
  }

  function renderCopyText() {
    if (copy && value) {
      return (
        <>
          {confirmation ? (
            <span className="textinput-action-button body-2 text-text-green pr-1">
              {t('icon.copied')}
            </span>
          ) : (
            <Button
              variant="icon"
              type="button"
              onClick={copyTextFromInput}
              className="textinput-action-button"
            >
              <i className="icon-copy text-lg inputIcon" />{' '}
            </Button>
          )}
        </>
      )
    }

    return null
  }

  function copyTextFromInput() {
    if (inputRef.current) {
      inputRef.current.select()
      document.execCommand('copy')
      setConfirmation(true)
    }
  }

  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
    const { value } = event.target
    onChange?.(value, event)
    setValue(value)
  }

  return (
    <div className="text-input-inner">
      <input
        {...rest}
        ref={inputRef}
        type={type ?? 'text'}
        className={clsx(
          className,
          { 'pr-6': clearAll },
          { 'text-input-error': validate === 'error' }
        )}
        onChange={handleChange}
      />

      {renderLengthMessage()}
      {renderClearButton()}
      {renderHelperText()}
      {renderCopyText()}
    </div>
  )
}

export default DefaultTextInput
