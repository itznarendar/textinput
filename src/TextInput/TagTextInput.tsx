import React from 'react'
import { InputPrtext } from './api'
import ReactTagInput from '@pathofdev/react-tag-input'
import '@pathofdev/react-tag-input/build/index.css'

function TagTextInput({
  defaultValue,
  onChange,
  placeholder,
  readOnly,
  disabled,
}: InputPrtext) {
  const [tags, setTags] = React.useState(
    (defaultValue as string)?.length
      ? (defaultValue as string)
          .replace(/^"|"$/g, '')
          .split('","')
          .map(i => i.replace(/\\"/g, '"'))
      : []
  )

  function handleChange(newTags:any) {
    setTags(newTags)
    onChange?.(newTags.map((i:any) => `"` + i.replace(`"`, `\\"`) + `"`).join(`,`))
  }

  return (
    <div className="text-input-outer">
      <ReactTagInput
        tags={tags}
        onChange={newTags => handleChange(newTags)}
        editable={!disabled}
        placeholder={placeholder}
        readOnly={readOnly}
      />
    </div>
  )
}

export default TagTextInput
