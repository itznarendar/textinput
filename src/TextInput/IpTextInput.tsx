import React, { useCallback, useReducer } from 'react'
import DefaultTextInput from './DefaultTextInput'
import { IpAddress, InputPrtext } from './api'

const reducer = <T extends object>(
  previousState: T = {} as T,
  updatedState: Partial<T> = {}
): T => {
  return { ...previousState, ...updatedState }
}
export const useSetState = <T extends object>(
  initialState: T = {} as T
): [T, (updatedState: Partial<T>) => void] => {
  const [state, dispatch] = useReducer<React.Reducer<T, Partial<T>>>(
    reducer,
    initialState
  )

  const setState = useCallback((updatedState: Partial<T>) => {
    dispatch(updatedState)
  }, [])
  return [state, setState]
}
function IpTextInput({ defaultValue, onChange }: InputPrtext) {
  const [ip, setIp] = useSetState<IpAddress>(
    getInitialIp(defaultValue as string)
  )

  // Ip valid value: [0, 255]
  function handleBlur(event: React.FocusEvent<HTMLInputElement>) {
    const { value } = event.target
    let validValue = value

    if (Number.parseInt(value) > 255) {
      validValue = '255'
    } else if (Number.parseInt(value) < 0) {
      validValue = '0'
    }

    if (validValue !== value) {
      onChange?.(getIp({ ...ip, [event.target.name]: validValue }))
      setIp({ [event.target.name]: validValue })
    }
  }

  function handleSubIpChange(
    value: string,
    event?: React.ChangeEvent<HTMLInputElement>
  ) {
    if (event) {
      onChange?.(getIp({ ...ip, [event.target.name]: value }))
      setIp({ [event.target.name]: value })
    }
  }

  function getDefaultPrtext(name: keyof IpAddress) {
    return {
      type: 'number',
      name,
      min: 0,
      max: 255,
      value: ip[name],
      onChange: handleSubIpChange,
      onBlur: handleBlur,
      className: 'border-0 text-center text',
    }
  }

  const classes = 'w-1/4 inline-block position-relative dot'

  return (
    <div className="text-input-outer">
      <div className={classes}>
        <DefaultTextInput {...getDefaultPrtext('first')} />
      </div>
      <div className={classes}>
        <DefaultTextInput {...getDefaultPrtext('second')} />
      </div>
      <div className={classes}>
        <DefaultTextInput {...getDefaultPrtext('third')} />
      </div>
      <div className="w-1/4 inline-block">
        <DefaultTextInput {...getDefaultPrtext('fourth')} />
      </div>
    </div>
  )
}

function getInitialIp(defaultValue: string) {
  const [first, second, third, fourth] =
    defaultValue?.split('.') ?? '...'.split('.')

  return { first, second, third, fourth }
}

function getIp(ip: IpAddress) {
  return ip.first || ip.second || ip.third || ip.fourth
    ? `${ip.first}.${ip.second}.${ip.third}.${ip.fourth}`
    : ''
}

export default IpTextInput
