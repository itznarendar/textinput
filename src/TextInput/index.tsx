import React from 'react'
import clsx from 'clsx'
import DefaultTextInput from './DefaultTextInput'
import IpTextInput from './IpTextInput'
import { InputPrtext, LabelPrtext } from './api'
import TagTextInput from './TagTextInput'

const TextInput: React.FC<InputPrtext> = ({
  type,
  label,
  clearAll,
  validate,
  maxLength,
  className,
  children,
  errors,
  ref,
  ...rest
}: InputPrtext) => {
  const { id } = rest
  const defaultInputPrtext = { clearAll, validate, maxLength, ref }
  const labelPrtext = { label, inputId: id, validate, required: rest.required }
  const contentClass = clsx('whitespace-nowrap', 'text-text-input-content')

  return (
    <div className={clsx('text-text-input body-2', className)} tabIndex={-1}>
      <div className="flex flex-row justify-between">
        {label ? <Label {...labelPrtext} /> : null}
        {children ? <div className={contentClass}>{children}</div> : null}
      </div>
      {type === 'ip' ? (
        <IpTextInput {...rest} />
      ) : type === 'tag' ? (
        <TagTextInput {...rest} />
      ) : (
        <DefaultTextInput {...rest} {...defaultInputPrtext} type={type} />
      )}
      {errors && <div className="body-3 text-text-red pt-1">{errors}</div>}
    </div>
  )
}

function Label({ label, inputId, validate, required }: LabelPrtext) {
  return (
    <label
      id={inputId ? `${inputId}-label` : undefined}
      htmlFor={inputId}
      className={clsx(
        { 'text-label-required': required },
        {
          'label-error': validate === 'error',
        },
        'body-3',
        'text-text-input-label'
      )}
    >
      {label}
    </label>
  )
}

export default TextInput
