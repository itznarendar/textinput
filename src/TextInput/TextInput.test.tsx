import * as React from 'react'
import renderer from 'react-test-renderer'
import TextInput from './index'

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: any) => key }),
}))

test('Text Input with max length', () => {
  const component = renderer
    .create(
      <TextInput
        type="text"
        id="b279e1b0-1d51-11ea-b576-0b857b03a511"
        label="Text Input with max length"
        placeholder="Enter text here"
        maxLength={10}
      />
    )
    .toJSON()
  expect(component).toMatchSnapshot()
})

test('Required Text Input', () => {
  const component = renderer
    .create(
      <TextInput
        type="text"
        id="e20ea460-1d24-11ea-b576-0b857b03a511"
        label="Required Text Input"
        placeholder="Enter text here"
        required
      />
    )
    .toJSON()
  expect(component).toMatchSnapshot()
})

test('Password input with pattern validation', () => {
  const component = renderer
    .create(
      <TextInput
        type="password"
        id="7e60c050-2d8d-11ea-beb5-b512248c9f49"
        label="Password Input"
        placeholder="Password must contain at least one number and one uppercase and lowercase letter, and at least 8 or more characters"
        pattern="(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,}"
        required
      />
    )
    .toJSON()
  expect(component).toMatchSnapshot()
})

test('Number Input in increments of 5', () => {
  const component = renderer
    .create(
      <TextInput
        type="number"
        id="c4eec400-1d51-11ea-b576-0b857b03a511"
        label="Number Input in increments of 5"
        placeholder="Enter a number here"
        step="5"
      />
    )
    .toJSON()
  expect(component).toMatchSnapshot()
})

test('Number Input in increments of 5', () => {
  const component = renderer
    .create(
      <TextInput
        type="number"
        id="c4eec400-1d51-11ea-b576-0b857b03a511"
        label="Number Input in increments of 5"
        placeholder="Enter a number here"
        step="5"
      />
    )
    .toJSON()
  expect(component).toMatchSnapshot()
})

test('Number Input with min and max', () => {
  const component = renderer
    .create(
      <TextInput
        type="number"
        id="da77eb30-1d51-11ea-b576-0b857b03a511"
        label="Number Input with min and max"
        placeholder="Enter a number between 1 and 10"
        min="1"
        max="10"
      />
    )
    .toJSON()
  expect(component).toMatchSnapshot()
})

test('Email Input', () => {
  const component = renderer
    .create(
      <TextInput
        type="email"
        id="76487d00-1d51-11ea-b576-0b857b03a511"
        label="Email Input"
        placeholder="Enter email here"
        required
      />
    )
    .toJSON()
  expect(component).toMatchSnapshot()
})

test('Telephone Input', () => {
  const component = renderer
    .create(
      <TextInput
        type="tel"
        id="f12ea070-1d52-11ea-b576-0b857b03a511"
        label="Mobile Number"
        placeholder="Enter phone number here in xxx-xxx-xxxx format"
        pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
      />
    )
    .toJSON()
  expect(component).toMatchSnapshot()
})

test('Date Input', () => {
  const component = renderer
    .create(
      <TextInput
        type="date"
        id="7c874de0-1d51-11ea-b576-0b857b03a511"
        label="Date Input"
        placeholder="Enter date here"
      />
    )
    .toJSON()
  expect(component).toMatchSnapshot()
})

test('Time Input', () => {
  const component = renderer
    .create(
      <TextInput
        type="time"
        id="9690bb20-1d53-11ea-b576-0b857b03a511"
        label="Time Input"
        placeholder="Enter time here"
      />
    )
    .toJSON()
  expect(component).toMatchSnapshot()
})

test('clean all button', async () => {
  const component = renderer
    .create(
      <TextInput
        type="time"
        id="9690bb20-1d53-11ea-b576-0b857b03a511"
        label="Time Input"
        placeholder="Enter time here"
        clearAll={true}
      />
    )
    .toJSON()
  expect(component).toMatchSnapshot()
})

test('validate in the input', async () => {
  const component = renderer
    .create(
      <TextInput
        type="time"
        id="9690bb20-1d53-11ea-b576-0b857b03a511"
        label="Time Input"
        placeholder="Enter time here"
        validate={'error'}
      />
    )
    .toJSON()
  expect(component).toMatchSnapshot()
})

test('validate in the input', async () => {
  const component = renderer
    .create(
      <TextInput
        type="time"
        id="9690bb20-1d53-11ea-b576-0b857b03a511"
        label="Time Input"
        placeholder="Enter time here"
        validate="ok"
      />
    )
    .toJSON()
  expect(component).toMatchSnapshot()
})
