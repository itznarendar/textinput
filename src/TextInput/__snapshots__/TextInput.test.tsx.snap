// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Input 1`] = `
<div
  className="text-text-input body-2"
  tabIndex={-1}
>
  <div
    className="flex flex-row justify-between"
  >
    <label
      className="body-3 text-text-input-label"
      htmlFor="7c874de0-1d51-11ea-b576-0b857b03a511"
      id="7c874de0-1d51-11ea-b576-0b857b03a511-label"
    >
      Date Input
    </label>
  </div>
  <div
    className="text-input-inner"
  >
    <input
      className=""
      id="7c874de0-1d51-11ea-b576-0b857b03a511"
      onChange={[Function]}
      placeholder="Enter date here"
      type="date"
    />
  </div>
</div>
`;

exports[`Email Input 1`] = `
<div
  className="text-text-input body-2"
  tabIndex={-1}
>
  <div
    className="flex flex-row justify-between"
  >
    <label
      className="text-label-required body-3 text-text-input-label"
      htmlFor="76487d00-1d51-11ea-b576-0b857b03a511"
      id="76487d00-1d51-11ea-b576-0b857b03a511-label"
    >
      Email Input
    </label>
  </div>
  <div
    className="text-input-inner"
  >
    <input
      className=""
      id="76487d00-1d51-11ea-b576-0b857b03a511"
      onChange={[Function]}
      placeholder="Enter email here"
      required={true}
      type="email"
    />
  </div>
</div>
`;

exports[`Number Input in increments of 5 1`] = `
<div
  className="text-text-input body-2"
  tabIndex={-1}
>
  <div
    className="flex flex-row justify-between"
  >
    <label
      className="body-3 text-text-input-label"
      htmlFor="c4eec400-1d51-11ea-b576-0b857b03a511"
      id="c4eec400-1d51-11ea-b576-0b857b03a511-label"
    >
      Number Input in increments of 5
    </label>
  </div>
  <div
    className="text-input-inner"
  >
    <input
      className=""
      id="c4eec400-1d51-11ea-b576-0b857b03a511"
      onChange={[Function]}
      placeholder="Enter a number here"
      step="5"
      type="number"
    />
  </div>
</div>
`;

exports[`Number Input in increments of 5 2`] = `
<div
  className="text-text-input body-2"
  tabIndex={-1}
>
  <div
    className="flex flex-row justify-between"
  >
    <label
      className="body-3 text-text-input-label"
      htmlFor="c4eec400-1d51-11ea-b576-0b857b03a511"
      id="c4eec400-1d51-11ea-b576-0b857b03a511-label"
    >
      Number Input in increments of 5
    </label>
  </div>
  <div
    className="text-input-inner"
  >
    <input
      className=""
      id="c4eec400-1d51-11ea-b576-0b857b03a511"
      onChange={[Function]}
      placeholder="Enter a number here"
      step="5"
      type="number"
    />
  </div>
</div>
`;

exports[`Number Input with min and max 1`] = `
<div
  className="text-text-input body-2"
  tabIndex={-1}
>
  <div
    className="flex flex-row justify-between"
  >
    <label
      className="body-3 text-text-input-label"
      htmlFor="da77eb30-1d51-11ea-b576-0b857b03a511"
      id="da77eb30-1d51-11ea-b576-0b857b03a511-label"
    >
      Number Input with min and max
    </label>
  </div>
  <div
    className="text-input-inner"
  >
    <input
      className=""
      id="da77eb30-1d51-11ea-b576-0b857b03a511"
      max="10"
      min="1"
      onChange={[Function]}
      placeholder="Enter a number between 1 and 10"
      type="number"
    />
  </div>
</div>
`;

exports[`Password input with pattern validation 1`] = `
<div
  className="text-text-input body-2"
  tabIndex={-1}
>
  <div
    className="flex flex-row justify-between"
  >
    <label
      className="text-label-required body-3 text-text-input-label"
      htmlFor="7e60c050-2d8d-11ea-beb5-b512248c9f49"
      id="7e60c050-2d8d-11ea-beb5-b512248c9f49-label"
    >
      Password Input
    </label>
  </div>
  <div
    className="text-input-inner"
  >
    <input
      className=""
      id="7e60c050-2d8d-11ea-beb5-b512248c9f49"
      onChange={[Function]}
      pattern="(?=.*\\\\d)(?=.*[a-z])(?=.*[A-Z]).{8,}"
      placeholder="Password must contain at least one number and one uppercase and lowercase letter, and at least 8 or more characters"
      required={true}
      type="password"
    />
  </div>
</div>
`;

exports[`Required Text Input 1`] = `
<div
  className="text-text-input body-2"
  tabIndex={-1}
>
  <div
    className="flex flex-row justify-between"
  >
    <label
      className="text-label-required body-3 text-text-input-label"
      htmlFor="e20ea460-1d24-11ea-b576-0b857b03a511"
      id="e20ea460-1d24-11ea-b576-0b857b03a511-label"
    >
      Required Text Input
    </label>
  </div>
  <div
    className="text-input-inner"
  >
    <input
      className=""
      id="e20ea460-1d24-11ea-b576-0b857b03a511"
      onChange={[Function]}
      placeholder="Enter text here"
      required={true}
      type="text"
    />
  </div>
</div>
`;

exports[`Telephone Input 1`] = `
<div
  className="text-text-input body-2"
  tabIndex={-1}
>
  <div
    className="flex flex-row justify-between"
  >
    <label
      className="body-3 text-text-input-label"
      htmlFor="f12ea070-1d52-11ea-b576-0b857b03a511"
      id="f12ea070-1d52-11ea-b576-0b857b03a511-label"
    >
      Mobile Number
    </label>
  </div>
  <div
    className="text-input-inner"
  >
    <input
      className=""
      id="f12ea070-1d52-11ea-b576-0b857b03a511"
      onChange={[Function]}
      pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
      placeholder="Enter phone number here in xxx-xxx-xxxx format"
      type="tel"
    />
  </div>
</div>
`;

exports[`Text Input with max length 1`] = `
<div
  className="text-text-input body-2"
  tabIndex={-1}
>
  <div
    className="flex flex-row justify-between"
  >
    <label
      className="body-3 text-text-input-label"
      htmlFor="b279e1b0-1d51-11ea-b576-0b857b03a511"
      id="b279e1b0-1d51-11ea-b576-0b857b03a511-label"
    >
      Text Input with max length
    </label>
  </div>
  <div
    className="text-input-inner"
  >
    <input
      className=""
      id="b279e1b0-1d51-11ea-b576-0b857b03a511"
      onChange={[Function]}
      placeholder="Enter text here"
      type="text"
    />
  </div>
</div>
`;

exports[`Time Input 1`] = `
<div
  className="text-text-input body-2"
  tabIndex={-1}
>
  <div
    className="flex flex-row justify-between"
  >
    <label
      className="body-3 text-text-input-label"
      htmlFor="9690bb20-1d53-11ea-b576-0b857b03a511"
      id="9690bb20-1d53-11ea-b576-0b857b03a511-label"
    >
      Time Input
    </label>
  </div>
  <div
    className="text-input-inner"
  >
    <input
      className=""
      id="9690bb20-1d53-11ea-b576-0b857b03a511"
      onChange={[Function]}
      placeholder="Enter time here"
      type="time"
    />
  </div>
</div>
`;

exports[`clean all button 1`] = `
<div
  className="text-text-input body-2"
  tabIndex={-1}
>
  <div
    className="flex flex-row justify-between"
  >
    <label
      className="body-3 text-text-input-label"
      htmlFor="9690bb20-1d53-11ea-b576-0b857b03a511"
      id="9690bb20-1d53-11ea-b576-0b857b03a511-label"
    >
      Time Input
    </label>
  </div>
  <div
    className="text-input-inner"
  >
    <input
      className="pr-6"
      id="9690bb20-1d53-11ea-b576-0b857b03a511"
      onChange={[Function]}
      placeholder="Enter time here"
      type="time"
    />
  </div>
</div>
`;

exports[`validate in the input 1`] = `
<div
  className="text-text-input body-2"
  tabIndex={-1}
>
  <div
    className="flex flex-row justify-between"
  >
    <label
      className="label-error body-3 text-text-input-label"
      htmlFor="9690bb20-1d53-11ea-b576-0b857b03a511"
      id="9690bb20-1d53-11ea-b576-0b857b03a511-label"
    >
      Time Input
    </label>
  </div>
  <div
    className="text-input-inner"
  >
    <input
      className="text-input-error"
      id="9690bb20-1d53-11ea-b576-0b857b03a511"
      onChange={[Function]}
      placeholder="Enter time here"
      type="time"
    />
    <i
      className="icon-error-circle text-lg validateIcon text-text-red"
    />
  </div>
</div>
`;

exports[`validate in the input 2`] = `
<div
  className="text-text-input body-2"
  tabIndex={-1}
>
  <div
    className="flex flex-row justify-between"
  >
    <label
      className="body-3 text-text-input-label"
      htmlFor="9690bb20-1d53-11ea-b576-0b857b03a511"
      id="9690bb20-1d53-11ea-b576-0b857b03a511-label"
    >
      Time Input
    </label>
  </div>
  <div
    className="text-input-inner"
  >
    <input
      className=""
      id="9690bb20-1d53-11ea-b576-0b857b03a511"
      onChange={[Function]}
      placeholder="Enter time here"
      type="time"
    />
    <i
      className="icon-check text-lg validateIcon text-text-green"
    />
  </div>
</div>
`;
